### ISSUE TITLE
ComboBox will bolck close all window   WinForm .Net Framework 4.8

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/winforms#7623]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/ComboBox-will-bolck-close-all-window---W/10122568)._

---
[severity:It bothers me. A fix would be nice]
Hi support.
Close all windows( taskbar -> right click icon -> close all window) can't work, with a asterisk sound when closing the form. 
The cause is the property 'AutoCompleteMode' are set not equal to None.

```
this.comboBox.AutoCompleteMode = System.Windows.Forms.AutoCompleteMode.Suggest
```
We discover that this control may not be disposed properly. A win32 error (code:1400) occurred when disposing the form.

---
#### Feedback Bot on 8/17/2022, 01:54 PM: 

(private comment, text removed)
#### Ashley Li [MSFT] on 8/18/2022, 02:53 PM: 

(private comment, text removed)
#### Luke Lu on 8/18/2022, 05:08 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by dreddy-work** [comment]
@Tanya-Solyanik/ @vladimir-krestov, do you guys recall any work around Combo Box for .NET framework 4.8 that could have caused this?

**Comment by vladimir-krestov** [comment]
I checked my and the team previous codeflows related to ComboBox, we changed an accessibility part only, so I don't remember any changes related to this bug.

**Comment by kirsan31** [comment]
Some more info on this.
1. `ComboBox` is out of business here 😁 You can delete it from child form and will get same behavior.
2. As @Ashley-Li mentioned the bug exists only if you open 1 or 2 child forms. If more - all will close.
3. If you open child form and normally close it (with cross), then next child form opening will not have this bug.

**Comment by merriemcgaw** [comment]
@Olina-Zhang can your team verify what we're seeing in .NET 8 and if something weird is still happening please rephrase the title of this to reflect what we see.

**Comment by Olina-Zhang** [comment]
@merriemcgaw it also repro in .Net 8 and the latest .Net 9 builds. And I try to create a simple application, form1 with button1 and empty form2. It has following code, also repro it.

```
private void button1_Click(object sender, EventArgs e)
{
    Form2 form2 = new Form2();
    form2.ShowDialog();
}

```
![Form_ClosedIssue](https://github.com/dotnet/winforms/assets/26474449/7a7c4f52-cc75-40f5-9298-3eb62c2ed7cc)

**Comment by elachlan** [comment]
I did a quick test and clicking "Close All Windows" will send a `WM_SYSCOMMAND` message via `WndProc` which will then trigger `WM_CLOSE`.

On the first run of the "Close All Windows", it will trigger `WM_SYSCOMMAND` once and then the "error sound" will play when `Dispose` is hit in `WmClose`, which ends up calling `NativeWindows.DestroyHandle` which in turn calls `PInvoke.DestroyWindow(HWND)` (sound plays on the PInvoke call).

On the second it will trigger `WM_SYSCOMMAND` twice (one after the other) and no sound is played.

I think this might be an operating system bug?

@Olina-Zhang Maybe try a MFC app and see if it suffers from the same problem?

**Comment by elachlan** [comment]
> This issue is reproduced from .NET Framework 4.8, but it cannot repro in .NET Framework 4.6.2~4.7.2.

So that rules out an OS issue, I guess?

**Comment by Olina-Zhang** [comment]
> @Olina-Zhang Maybe try a MFC app and see if it suffers from the same problem?

Verified it in a MFC app I created, CloseWindow by "taskbar -> right click app icon -> Close  Window" doesn't work when second dialog opening.
![MFC_CloseWindow](https://github.com/dotnet/winforms/assets/26474449/b1601d31-2cbc-4b64-88f5-3ffef5c2a4b4)

**Comment by merriemcgaw** [comment]
Interesting @Olina-Zhang that tends to point more to some Win32 API issues. Can you check a Win10 OS and see if the OS version is the key to whether MFC (and .NET based) works? 

If this is us (but with MFC doing the same thing I doubt it) then I'd want a fix in .NET 9 if possible. If it's Win10 vs Win11, I'd be interested to know if the later .NET Framework versions on the older Win10 builds exhibit the same problem.

**Comment by Olina-Zhang** [comment]
@merriemcgaw I tested it on Win10 22H2 OS in MFC application, it is same as above Winforms behavior(GIF) I attached

**Comment by elachlan** [comment]
@Olina-Zhang has an internal ticket been raised? I Imagine that is the next step.

**Comment by Olina-Zhang** [comment]
@elachlan No, I am not sure what is the exactly path of Windows team about this issue. @merriemcgaw can you point to me if it needs?

**Comment by merriemcgaw** [comment]
I would just use Windows Feedback I think. Attach the videos from both UI stacks. Someone will route it to the right place. I too don't know which of the many Windows teams it would be. If it were common controls, or common dialog, or even Win32 I could find it.

**Comment by Olina-Zhang** [comment]
Thanks @merriemcgaw, I have submitted a feedback to Windows by Windows feedback. Closing this issue now.
